v2/spanner-common/src/main/java/com/google/cloud/teleport/v2/spanner/migrations/schema/SessionBasedMapper.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.spanner.migrations.schema;
+
+import com.google.cloud.teleport.v2.spanner.ddl.Column;
+import com.google.cloud.teleport.v2.spanner.ddl.Ddl;
+import com.google.cloud.teleport.v2.spanner.ddl.Table;
+import com.google.cloud.teleport.v2.spanner.type.Type;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+
+public class SessionBasedMapper implements ISchemaMapper {
+
+  Ddl ddl;
+
+  Schema schema;
+
+  public SessionBasedMapper(Schema schema, Ddl ddl) {
```

Comment:
It is named SessionBasedMapper , but there is nothing that reflects a session file. Does it make sense to pass the session file path here instead, rather than Schema?

Reply:
We could do that, or rename this to SchemaBasedMapper?

Reply:
Depends on if the session file or the Schema object will be used elsewhere other than this mapper. If Schema object is not needed elsewhere in the code, then no need to expose it and only a session file can be taken here as input.

Reply:
Makes sense